procrastinate/manager.py

```diff
@@ -224,21 +224,23 @@ async def finish_job_by_id_async(
     async def retry_job(
         self,
         job: jobs.Job,
-        retry_at: datetime.datetime,
+        retry_at: Optional[datetime.datetime] = None,
     ) -> None:
         """
         Indicates that a job should be retried later.
 
         Parameters
         ----------
         job : `jobs.Job`
-        retry_at : ``datetime.datetime``
+        retry_at : ``Optional[datetime.datetime]``
             If set at present time or in the past, the job may be retried immediately.
             Otherwise, the job will be retried no sooner than this date & time.
-            Should be timezone-aware (even if UTC)
+            Should be timezone-aware (even if UTC). Defaults to present time.
         """
-        assert job.id  # TODO remove this
-        await self.retry_job_by_id_async(job_id=job.id, retry_at=retry_at)
+        assert job.id  # TODO remove this await
```

Comment:
wait what ?

Reply:
bad copy&paste I guess, thanks for the sharp eyes! Now fixed.